token/program-2022/src/processor.rs

```diff
@@ -149,6 +149,7 @@ impl Processor {
         };
 
         account.pack_base();
+        account.init_account_type()?;
```

Comment:
@joncinque just wanted to point out that I missed this in #2743
Maybe it doesn't matter because the scope of use is so small, but would it make sense to have successful extension inits do this under the hood?

Reply:
Yeah I noticed, did the end-to-end test catch this?

> Maybe it doesn't matter because the scope of use is so small, but would it make sense to have successful extension inits do this under the hood?

With mints, we initialize all extensions *and then* initialize the base + account type, so I'd prefer to keep that flow for accounts too, even though it's all done within the same instruction.  But I don't feel super strongly about this.

Reply:
> did the end-to-end test catch this or the processor-level test?

The end-to-end, since it depended on full-fledged get-/set-return-data functionality to detect.

Okay. Could maybe be bundled with base init, then. But I'll drop it for now, and readdress if it crops up again later.

Reply:
> Could maybe be bundled with base init, then.

Good call, that'll probably make the most sense